src/carbon-cycle-solver.cpp

```diff
@@ -15,7 +15,12 @@
 
 #include <math.h>
 #include <string>
+
+// some boost headers generate warnings under clang; not our problem, ignore
+#pragma clang diagnostic push
```

Comment:
Does the pragma call silence the warnings?

Reply:
`#pragma` is an open-ended extension to the C++ preprocessor through which each compiler can implement custom functionality.

Here, the first `#pragma` says "I'm about to send you a command to push onto your stack" (where the top of the stack represents the active settings).

Second command says "ignore all warnings"

Then the `#include`

And then finally we tell clang to pop off (remove) this instruction, restoring previous behavior.